components/payloads/cu_sensor_payloads/src/lib.rs

```diff
@@ -106,6 +110,150 @@ impl PointCloud {
     }
 }
 
+fn color_type_to_num(color_type: ColorType) -> u16 {
+    match color_type {
+        ColorType::L8 => 0,
+        ColorType::La8 => 1,
+        ColorType::Rgb8 => 2,
+        ColorType::Rgba8 => 3,
+        ColorType::L16 => 4,
+        ColorType::La16 => 5,
+        ColorType::Rgb16 => 6,
+        ColorType::Rgba16 => 7,
+        ColorType::Rgb32F => 8,
+        ColorType::Rgba32F => 9,
+        _ => unreachable!("color_type: {:?} is unsupported", color_type),
+    }
+}
+fn color_type_from_num(num: u16) -> ColorType {
+    match num {
+        0 => ColorType::L8,
+        1 => ColorType::La8,
+        2 => ColorType::Rgb8,
+        3 => ColorType::Rgba8,
+        4 => ColorType::L16,
+        5 => ColorType::La16,
+        6 => ColorType::Rgb16,
+        7 => ColorType::Rgba16,
+        8 => ColorType::Rgb32F,
+        9 => ColorType::Rgba32F,
+        _ => unreachable!("color_id: {:?} is unsupported", num),
+    }
+}
+
+#[derive(Default, Clone, PartialEq, Debug)]
+pub struct CuImage(DynamicImage);
+
+impl Deref for CuImage {
+    type Target = DynamicImage;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for CuImage {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl CuImage {
+    pub fn new(img: DynamicImage) -> Self {
+        Self(img)
+    }
+}
+
+impl Encode for CuImage {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        // println!("enc w: {} h: {}", self.width(), self.height());
+        Encode::encode(&self.width(), encoder)?;
```

Comment:
Thanks for the PR, it helps settling on something!

I think we want all that (ie width, height, stride, pixel type) to be static, ie in the type itself Cuimage<1920, 1200, RGBA, 1920>.
Like that for the images in (main) memory we know their size (they implement Sized) and we can simply preallocate them like we do for the rest.
Now (see the discussion on discord) we might want also a GPU / NPU / NUMA type for images that are no in the CopperList.

Reply:
For the first part, I think I am wrong: looking at some hw, the stride can change (I see some camera with dynamic cropping for example)